daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/walker/AbstractDSOMWalker.scala

```diff
@@ -35,6 +35,7 @@ trait CommonContextView {
 trait TermView extends CommonContextView {
   def isArray: Boolean
   def isOptional: Boolean
+  final def isArrayOrOptional: Boolean = isArray || isOptional
```

Comment:
Ask your IDEA if there are any uses of` this `isArrayOrOptional` function.  Code coverage says this function is not covered by tests, which implies that this function is never used anywhere else (if so, we should remove it).

Reply:
This was added as a potentially-useful helper in the future, but can be removed as it is currently not used.